execution_engine/src/core/runtime/mod.rs

```diff
@@ -1080,6 +1080,15 @@ where
             }
         };
 
+        #[cfg(feature = "test-support")]
+        if let wasmi::Error::Trap(Trap::Code(TrapCode::Unreachable)) = &error {
```

Comment:
More detailed debug info added in 8ff446d6f66cc94f5f21e613bdeef152f76487b3.
However, for consistency, I print `current_wasm_instance_stack_height` for every error.

Reply:
@mpapierski as per discussion, this solution has some drawbacks, in particular:

For recursive calls, as observed in `regression_20211110.rs` test, the **single** `RuntimeStackOverflow` will cause **multiple** error messages to be printed (one for each runtime being unwinded from the stack), like so:
```
Wasm execution error: Trap: Host(RuntimeStackOverflow)
current_wasm_instance_stack_height=29
current_runtime_stack_height=11
max_runtime_stack_height=12

--- frame 10 ---
--- frame 9 ---
...
--- frame 0 ---
[.....more details cut for legibility.....]

Wasm execution error: Trap: Host(RuntimeStackOverflow)
current_wasm_instance_stack_height=29
current_runtime_stack_height=10
max_runtime_stack_height=12

--- frame 9 ---
...
--- frame 0 ---
[.....more details cut for legibility.....]

Wasm execution error: Trap: Host(RuntimeStackOverflow)
current_wasm_instance_stack_height=29
current_runtime_stack_height=9
max_runtime_stack_height=12

--- frame 9 ---
...
--- frame 0 ---
[.....more details cut for legibility.....]
```

Additionally, in the initial implementation which was only referencing the `Trap::Unreachable`, we were able to only analyze the error if the call to `invoke_export` failed, because there was no chance for this error to happen earlier in the flow.

Now, if we're trying to display debug info for other errors, we need to cover more places in `execute_contract()`, for example, we may fail at `stack.push(call_stack_element)?;` even before runtime is created.

It would be best to dump the error info in a single place, probably within `Executor::exec()`, but the problem is that internally the `fn execute_contract()` erases info about `wasmi::Error` and yields it as a String representation `Interpreter(String)`, hence we cannot act on a specific error type upper in the call-stack (unless we want to compare strings, but we don't).

Therefore I will:
* Prepare another PR with just the simple, initial debug info (as described in [the ticket](https://github.com/casper-network/casper-node/issues/4003))
* Close this PR
* Request a ticket to build on top of this closed PR and add the ability to provide users with the more detailed dump as was proposed in the comment [here](https://github.com/casper-network/casper-node/pull/4102#discussion_r1251007300)